synapse/handlers/register.py

```diff
@@ -128,6 +128,15 @@ def check_username(self, localpart, guest_access_token=None, assigned_user_id=No
                     errcode=Codes.FORBIDDEN,
                 )
 
+        if guest_access_token is not None:
```

Comment:
I think this logic is reversed, the removed logic is:

```python
was_guess = guest_access_token is not None
if not was_guess:
    ...
```

So I think this should be `if guest_access_token is None`.

Reply:
Ugh, yes, thanks

Reply:
That explains why the sytests were all failing fwiw.